gaffer-as-a-service/gaas-rest/src/main/java/uk/gov/gchq/gaffer/gaas/auth/WebSecurityConfig.java

```diff
@@ -63,29 +68,47 @@ public AuthenticationManager authenticationManagerBean() throws Exception {
         return super.authenticationManagerBean();
     }
 
+
     @Override
-    protected void configure(final HttpSecurity httpSecurity) throws Exception {
-        // We don't need CSRF for this example
-        httpSecurity.csrf().disable()
-                // dont authenticate this particular request
-                .authorizeRequests()
-                .antMatchers("/auth", "/v2/api-docs", "/swagger-ui.html", "/swagger-ui/", "/swagger-ui/**", "/swagger-resources",
-                        "/swagger-resources/**", "/webjars/**", "/actuator/**")
-                .permitAll()
-                // all other requests need to be authenticated
-                .anyRequest()
-                .authenticated()
-                .and()
-                // make sure we use stateless session; session won't be used to
-                // store user's state.
-                .exceptionHandling()
-                .authenticationEntryPoint(jwtAuthenticationEntryPoint)
-                .and()
-                .sessionManagement()
-                .sessionCreationPolicy(SessionCreationPolicy.STATELESS);
-
-        // Add a filter to validate the tokens with every request
-        httpSecurity.addFilterBefore(jwtRequestFilter, UsernamePasswordAuthenticationFilter.class);
-        httpSecurity.cors();
+    protected void configure(final HttpSecurity http) throws Exception {
+        if (cognitoEnabled) {
+            http
+                    .authorizeRequests(authorizeRequests ->
+                            authorizeRequests
+                                    .antMatchers("/v2/api-docs", "/swagger-ui.html", "/swagger-ui/", "/swagger-ui/**", "/swagger-resources",
+                                            "/swagger-resources/**", "/webjars/**", "/actuator/**").hasAuthority("SCOPE_services:read")
+                                    .anyRequest().authenticated()
+                    )
+                    .oauth2ResourceServer(oauth2ResourceServer ->
+                            oauth2ResourceServer
+                                    .jwt(jwt ->
+                                            jwt.decoder(JwtDecoders.fromIssuerLocation(issuerUri))
+                                    )
+                    );
+        } else {
+            http.csrf().disable() // nosemgrep: java.spring.security.audit.spring-csrf-disabled.spring-csrf-disabled
```

Comment:
Perhaps a comment explaining why csrf is disabled here

Reply:
Sure

Reply:
Added one in let me know if you think it is sufficient or not